### ISSUE TITLE
[HOLD for payment 2024-07-22] [HOLD for payment 2024-07-17] [Report Fields] [External] implement the policy report field delete flow

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#43686]
[Main tracking issue].(https://github.com/Expensify/Expensify/issues/365155)

[Design doc link](https://docs.google.com/document/d/10b7pTVvgpB6_O8sRTKaCzZEVM3Ba6hxnIwyyQSSXCLY/edit). 

Implement the following section of the design doc: https://docs.google.com/document/d/10b7pTVvgpB6_O8sRTKaCzZEVM3Ba6hxnIwyyQSSXCLY/edit#bookmark=kix.53mwf06dzdi0

This might technically be on hold for https://github.com/Expensify/App/issues/43677 but I think Callstack can start on these in parallel using feature branch.

Include unit tests for the optimistic actions 

<details><summary>Issue Owner</summary>Current Issue Owner: @adelekennedy</details>

**Comment by waterim** [comment]
Hello, Im Artem from Callstack and would like to help with this issue

**Comment by mountiny** [comment]
I believe we are all done here
